File: Unit_Testing_C++/src/order.cpp
// order.cpp
#include "order.h"

Order::Order(int order_id, Customer customer, std::vector<std::pair<Product, int>> products)
    : order_id(order_id), customer(customer), products(products) {}

double Order::calculate_total() const {
    double total = 0.0;
    for (const auto& pair : products) {
        total += pair.first.get_price() * pair.second;
    }
    return total;
}

std::string Order::display_info() const {
    std::string product_list;
    for (const auto& pair : products) {
        product_list += pair.first.get_name() + " - $" + std::to_string(pair.first.get_price()) + ", Quantity: " + std::to_string(pair.second) + "\n";
    }
    return "Order ID: " + std::to_string(order_id) + "\nCustomer: " + customer.display_info() + "\nProducts:\n" + product_list + "\nTotal: $" + std::to_string(calculate_total());
}

Customer Order::get_customer() const {
    return customer;
}

std::vector<std::pair<Product, int>> Order::get_products() const {
    return products;
}

File: Unit_Testing_Python/src/order.py
# order.py
class Order:
    def __init__(self, order_id, customer, products):
        self.order_id = order_id
        self.customer = customer
        self.products = products

    def calculate_total(self):
        return sum(product.price * quantity for product, quantity in self.products)

    def display_info(self):
        product_list = "\n".join([f"{product.name} - ${product.price}, Quantity: {quantity}" for product, quantity in self.products])
        return f"Order ID: {self.order_id}\nCustomer: {self.customer.name}\nProducts:\n{product_list}\nTotal: ${self.calculate_total()}"
